src/powerquery-formatter/passes/preProcessParameter.ts

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as PQP from "@microsoft/powerquery-parser";
+
+import { Trace, TraceManager } from "@microsoft/powerquery-parser/lib/powerquery-parser/common/trace";
+import { Ast } from "@microsoft/powerquery-parser/lib/powerquery-parser/language";
+
+import { CommentCollectionMap, SerializeParameter, SerializeParameterMap } from "./commonTypes";
+import { FormatTraceConstant } from "../trace";
+import { getNodeScopeName } from "../themes";
+import { NodeKind } from "@microsoft/powerquery-parser/lib/powerquery-parser/language/ast/ast";
+
+interface PreProcessState extends PQP.Traverse.ITraversalState<SerializeParameterMap> {
+    nodeIdMapCollection: PQP.Parser.NodeIdMap.Collection;
+    commentCollectionMap: CommentCollectionMap;
+    cancellationToken: PQP.ICancellationToken | undefined;
+    currentNode: Ast.TNode;
+    visitedNodes: Ast.TNode[];
+    visitedNodeScopeNames: string[];
+    result: SerializeParameterMap;
+}
+
+type ParameterPreProcessor = (state: PreProcessState) => void;
+
+function parameterizeOneNodeInForceBlockMode(
+    currentNode: Ast.TNode,
+    serializeParameterMap: Map<number, SerializeParameter>,
+): void {
+    let maybeCurrentNodeSerializeParameter: SerializeParameter | undefined = serializeParameterMap.get(currentNode.id);
```

Comment:
nit: we're dropping the `maybe` prefix on `undefined`able variables.

Reply:
emmm, i just started to like it👻